In [1]:
import pandas as pd
import numpy as np

In [44]:
bank = pd.read_csv("bank_marketing.csv", index_col=(0))

In [45]:
bank

age          job  marital  education default  balance housing loan  \
0      41     services  married    unknown      no       88     yes   no   
1      56   technician  married  secondary      no     1938      no  yes   
2      30     services   single  secondary      no      245      no  yes   
3      34   management   single   tertiary      no     1396     yes   no   
4      29   technician   single  secondary      no      -13     yes   no   
...   ...          ...      ...        ...     ...      ...     ...  ...   
5576   48       admin.  married  secondary      no     2235     yes  yes   
5577   40  blue-collar  married    primary      no     -388     yes   no   
5578   27      student   single  secondary      no       91      no   no   
5579   46  blue-collar  married  secondary      no     1568     yes   no   
5580   39     services  married  secondary      no        0     yes   no   

        contact  day month  duration  campaign  pdays  previous poutcome  \
0      cellular   11   may       105         1    336         2  failure   
1      cellular   26   feb       229         1    192         4  success   
2      cellular    8   jul       187         2     -1         0  unknown   
3      cellular   17   jul       630         1     -1         0  unknown   
4      cellular   14   may       512         3     -1         0  unknown   
...         ...  ...   ...       ...       ...    ...       ...      ...   
5576    unknown   15   may       755         2     -1         0  unknown   
5577   cellular    4   may       195         3    340         1  failure   
5578  telephone    4   dec       157         6     95        37    other   
5579   cellular    2   jun       310         1     55         6    other   
5580   cellular    8   may       503         1     -1         0  unknown   

     deposit  
0         no  
1        yes  
2         no  
3         no  
4         no  
...      ...  
5576      no  
5577      no  
5578      no  
5579     yes  
5580      no  

[5581 rows x 17 columns]

In [46]:
bank.isnull().sum()

age          0
job          0
marital      0
education    0
default      0
balance      0
housing      0
loan         0
contact      0
day          0
month        0
duration     0
campaign     0
pdays        0
previous     0
poutcome     0
deposit      0
dtype: int64

In [47]:
x = bank[bank.duplicated(keep="first")]
x

Empty DataFrame
Columns: [age, job, marital, education, default, balance, housing, loan, contact, day, month, duration, campaign, pdays, previous, poutcome, deposit]
Index: []

### Average age of the clients those who have subscribed to deposit

In [25]:
age_df = pd.DataFrame(bank)
age_df = age_df[age_df["deposit"]=="yes"]
age_df

age          job   marital  education default  balance housing loan  \
1      56   technician   married  secondary      no     1938      no  yes   
5      26   unemployed    single  secondary      no      632      no   no   
6      59      retired   married    primary      no     2074      no   no   
8      61      retired   married  secondary      no      569      no   no   
9      35  blue-collar  divorced  secondary      no      336     yes   no   
...   ...          ...       ...        ...     ...      ...     ...  ...   
5570   66      retired   married   tertiary      no     1948      no   no   
5571   37     services    single  secondary      no      278     yes   no   
5573   45   technician    single  secondary      no      410     yes   no   
5574   47     services   married  secondary      no      405      no  yes   
5579   46  blue-collar   married  secondary      no     1568     yes   no   

       contact  day month  duration  campaign  pdays  previous poutcome  \
1     cellular   26   feb       229         1    192         4  success   
5     cellular   12   feb       218         1     -1         0  unknown   
6     cellular   21   dec       467         1    186         2  success   
8     cellular    4   aug       219         1     -1         0  unknown   
9     cellular   14   may       191         1    346         2  failure   
...        ...  ...   ...       ...       ...    ...       ...      ...   
5570  cellular   28   jan       216         1     91         4  success   
5571   unknown    3   jun      1045         3     -1         0  unknown   
5573   unknown   30   may       891         4     -1         0  unknown   
5574  cellular    8   aug       994         2     -1         0  unknown   
5579  cellular    2   jun       310         1     55         6    other   

     deposit  
1        yes  
5        yes  
6        yes  
8        yes  
9        yes  
...      ...  
5570     yes  
5571     yes  
5573     yes  
5574     yes  
5579     yes  

[2622 rows x 17 columns]

In [26]:
age_df["age"].describe()

count    2622.000000
mean       41.426773
std        13.560264
min        18.000000
25%        31.000000
50%        38.000000
75%        50.000000
max        93.000000
Name: age, dtype: float64

### Maximum number of contacts performed during the campaign for the clients who have not subscribed to deposit

In [24]:
ndepo = pd.DataFrame(bank)
ndepo = ndepo[ndepo["deposit"]=="no"]
ndepo

age          job  marital  education default  balance housing loan  \
0      41     services  married    unknown      no       88     yes   no   
2      30     services   single  secondary      no      245      no  yes   
3      34   management   single   tertiary      no     1396     yes   no   
4      29   technician   single  secondary      no      -13     yes   no   
7      46   management  married   tertiary      no        3      no   no   
...   ...          ...      ...        ...     ...      ...     ...  ...   
5575   31   technician  married  secondary      no      589     yes  yes   
5576   48       admin.  married  secondary      no     2235     yes  yes   
5577   40  blue-collar  married    primary      no     -388     yes   no   
5578   27      student   single  secondary      no       91      no   no   
5580   39     services  married  secondary      no        0     yes   no   

        contact  day month  duration  campaign  pdays  previous poutcome  \
0      cellular   11   may       105         1    336         2  failure   
2      cellular    8   jul       187         2     -1         0  unknown   
3      cellular   17   jul       630         1     -1         0  unknown   
4      cellular   14   may       512         3     -1         0  unknown   
7      cellular   18   aug        92         4     -1         0  unknown   
...         ...  ...   ...       ...       ...    ...       ...      ...   
5575    unknown   21   may       151         1     -1         0  unknown   
5576    unknown   15   may       755         2     -1         0  unknown   
5577   cellular    4   may       195         3    340         1  failure   
5578  telephone    4   dec       157         6     95        37    other   
5580   cellular    8   may       503         1     -1         0  unknown   

     deposit  
0         no  
2         no  
3         no  
4         no  
7         no  
...      ...  
5575      no  
5576      no  
5577      no  
5578      no  
5580      no  

[2959 rows x 17 columns]

In [27]:
ndepo["campaign"].describe()

count    2959.000000
mean        2.833728
std         3.308879
min         1.000000
25%         1.000000
50%         2.000000
75%         3.000000
max        63.000000
Name: campaign, dtype: float64

### Difference between the maximum balance (in euros) for the clients who have subscribed to deposit and for the clients who have not subscribed to the deposit

In [39]:
print("[Max Balance of Deposit = yes] : ", age_df["balance"].max())
print("[Max Balance of Deposit = no] : ",ndepo["balance"].max())
age_df["balance"].max()-ndepo["balance"].max()

[Max Balance of Deposit = yes] :  81204
[Max Balance of Deposit = no] :  56831


24373

### Unique job levels in the data and No of clients in management level

In [40]:
bank["job"].describe()

count           5581
unique            12
top       management
freq            1318
Name: job, dtype: object

##### 12 level of jobs and 1318 clients in management level 